sorc/build_ww3prepost.sh

```diff
@@ -1,131 +1,105 @@
-module use "${script_dir}/ufs_model.fd/modulefiles"
+source "${_HOMEgfs}/sorc/ufs_model.fd/tests/detect_machine.sh"
+source "${_HOMEgfs}/sorc/ufs_model.fd/tests/module-setup.sh"
+module use "${_HOMEgfs}/sorc/ufs_model.fd/modulefiles"
 module load "ufs_${MACHINE_ID}.intel"
 set -x
 
-#Set WW3 directory, switch, prep and post exes
-cd ufs_model.fd/WW3 || exit 1
-WW3_DIR=$( pwd -P )
+#Set WW3 directory
+cd "${_HOMEgfs}/sorc/ufs_model.fd/WW3" || exit 1
+WW3_DIR=$(pwd -P)
 export WW3_DIR
-export SWITCHFILE="${WW3_DIR}/${ww3switch}"
 
-# Build exes for prep jobs and post jobs:
-prep_exes="ww3_grid ww3_prep ww3_prnc ww3_grid"
-post_exes="ww3_outp ww3_outf ww3_outp ww3_gint ww3_ounf ww3_ounp ww3_grib"
+# Determine which switch to use
+if [[ "${PDLIB}" == "ON" ]]; then
+  ww3switch="model/bin/switch_meshcap_pdlib"
+  path_build="${WW3_DIR}/build/pdlib_ON"
+  path_install="${WW3_DIR}/install/pdlib_ON"
+else
+  ww3switch="model/bin/switch_meshcap"
+  path_build="${WW3_DIR}/build/pdlib_OFF"
+  path_install="${WW3_DIR}/install/pdlib_OFF"
+fi
+export SWITCHFILE="${WW3_DIR}/${ww3switch}"
 
 #create build directory:
-path_build="${WW3_DIR}/build_SHRD"
 [[ -d "${path_build}" ]] && rm -rf "${path_build}"
 mkdir -p "${path_build}" || exit 1
 cd "${path_build}" || exit 1
-echo "Forcing a SHRD build"
 
 buildswitch="${path_build}/switch"
 
-cat "${SWITCHFILE}" > "${path_build}/tempswitch"
-
-sed -e "s/DIST/SHRD/g"\
-    -e "s/OMPG / /g"\
-    -e "s/OMPH / /g"\
-    -e "s/MPIT / /g"\
-    -e "s/MPI / /g"\
-    -e "s/B4B / /g"\
-    -e "s/PDLIB / /g"\
-    -e "s/SCOTCH / /g"\
-    -e "s/METIS / /g"\
-    -e "s/NOGRB/NCEP2/g"\
-       "${path_build}/tempswitch" > "${path_build}/switch"
+cat "${SWITCHFILE}" >"${path_build}/tempswitch"
+
+sed -e "s/DIST/SHRD/g" \
+  -e "s/OMPG / /g" \
+  -e "s/OMPH / /g" \
+  -e "s/MPIT / /g" \
+  -e "s/MPI / /g" \
+  -e "s/B4B / /g" \
+  -e "s/PDLIB / /g" \
+  -e "s/SCOTCH / /g" \
+  -e "s/METIS / /g" \
+  -e "s/NOGRB/NCEP2/g" \
+  "${path_build}/tempswitch" >"${path_build}/switch"
 rm "${path_build}/tempswitch"
 
 echo "Switch file is ${buildswitch} with switches:"
 cat "${buildswitch}"
 
 #define cmake build options
-MAKE_OPT="-DCMAKE_INSTALL_PREFIX=install"
+MAKE_OPT="-DCMAKE_INSTALL_PREFIX=${path_install}"
 [[ ${BUILD_TYPE:-"Release"} = "Debug" ]] && MAKE_OPT+=" -DCMAKE_BUILD_TYPE=Debug"
 
 #Build executables:
 # shellcheck disable=SC2086
 cmake "${WW3_DIR}" -DSWITCH="${buildswitch}" ${MAKE_OPT}
 rc=$?
-if (( rc != 0 )); then
+if ((rc != 0)); then
   echo "Fatal error in cmake."
   exit "${rc}"
 fi
+
 make -j "${BUILD_JOBS:-8}"
 rc=$?
-if (( rc != 0 )); then
+if ((rc != 0)); then
   echo "Fatal error in make."
   exit "${rc}"
 fi
+
 make install
-if (( rc != 0 )); then
+rc=$?
+if ((rc != 0)); then
   echo "Fatal error in make install."
   exit "${rc}"
 fi
 
-# Copy to top-level exe directory
-for prog in ${prep_exes} ${post_exes}; do
-  cp "${path_build}/install/bin/${prog}" "${finalexecdir}/"
```

Comment:
Is there a reason that this was moved to the link file?    Is it normal to have to re-link after building? I thought everything just went where it was supposed to after rebuilding... seems like this might cause some unintentional errors (at least for this script, but maybe this wasn't ever supposed to be here?)

Reply:
yes.  This is inline with how ufs-weather-model, upp, gsi, etc are being handled.  
I don't think this is good/bad/preferred or anything like that.  
Since we need to build for a GFS and GEFS, this seemed to be
- consistent w/ other build processes
- dependent on pdlib, rather than gfs|gefs
This is not to say it cannot be updated in the future if a more appropriate `make install` can be executed in the WW3 cmake build that allows one to install the built executables with a `gfs_` or `gefs_` prefix.

Reply:
Thanks for the explanation @aerorahul  - I see now how this is consistent with the other build processes.  Thanks for making this consistent.